ironfist/src/cpp/combat/creatures.cpp

```diff
@@ -50,6 +52,17 @@ attributeNameTableEntry creatureAttributeNameTable[] = {
 	{"undead", UNDEAD}
 };
 
+struct secondaryCostIdxTableEntry { char * resource; int idx; };
+
+secondaryCostIdxTableEntry secondaryCostIdxTable[] = {
+	{"wood", 0},

```

Comment:
These integer constants are "magic numbers."

In the words of my TA, "The number 1 is not mercury. The number  is the number 1."

I'd like you to cultivate a habit of always looking for a place where these constants are already defined. In this case, the file HEROES2W.h taken from the reverse-engineering work defines an enum called RESOURCES with constants like RESOURCE_WOOD, etc. If this isn't already somewhere in the Ironfist project, you can add it, and then refer to those constants.

I think there should also be a table somewhere in the executable giving the names of resources by ID. You should reuse that table. Like, instead of having an entry be {"wood", 0}, have the entry be {gResourceNames[RESOURCE_WOOD], RESOURCE_WOOD}.